Sources/Subprocess/Subprocess+AsyncBytes.swift

```diff
@@ -9,167 +9,91 @@
         public struct Iterator: AsyncIteratorProtocol {
-
-            @inline(__always) static var bufferSize: Int {
-                16384
-            }
-
             public typealias Element = UInt8
-            @usableFromInline var buffer: _AsyncBytesBuffer
 
-            internal var byteBuffer: _AsyncBytesBuffer {
-                return buffer
+            private let fileDescriptor: FileDescriptor
+            private var buffer: [UInt8]
+            private var currentPosition: Int
+            private var finished: Bool
+
+            internal init(fileDescriptor: FileDescriptor) {
+                self.fileDescriptor = fileDescriptor
+                self.buffer = []
+                self.currentPosition = 0
+                self.finished = false
             }
 
-            internal init(file: FileDescriptor) {
-                buffer = _AsyncBytesBuffer(capacity: Iterator.bufferSize)
-                buffer.readFunction = { (buf) in
-                    buf.nextPointer = buf.baseAddress
-                    let capacity = buf.capacity
-                    let bufPtr = UnsafeMutableRawBufferPointer(start: buf.nextPointer, count: capacity)
-                    let readSize: Int = try await IOActor().read(from: file, into: bufPtr)
-                    buf.endPointer = buf.nextPointer + readSize
-                    return readSize
+            private mutating func reloadBufferAndNext() async throws -> UInt8? {
+                if self.finished {
+                    return nil
+                }
+                try Task.checkCancellation()
+                do {
+                    self.buffer = try await self.read(
+                        from: self.fileDescriptor,
+                        maxLength: AsyncBytes.bufferSize)
+                    self.currentPosition = 0
+                    if self.buffer.count < AsyncBytes.bufferSize {
+                        self.finished = true
+                    }
+                } catch {
+                    self.finished = true
+                    throw error
                 }
+                return try await self.next()
             }
 
-            @inlinable @inline(__always)
             public mutating func next() async throws -> UInt8? {
-                return try await buffer.next()
+                if currentPosition < buffer.count {
+                    let value = buffer[currentPosition]
+                    self.currentPosition += 1
+                    return value
+                }
+                return try await self.reloadBufferAndNext()
+            }
+
+            private func read(from fileDescriptor: FileDescriptor, maxLength: Int) async throws -> [UInt8] {
+                return try await withCheckedThrowingContinuation { continuation in
+                    DispatchIO.read(
+                        fromFileDescriptor: fileDescriptor.rawValue,
+                        maxLength: maxLength,
+                        runningHandlerOn: .main
```

Comment:
This isn't a good idea, this suddenly relies on the `DispatchQueue.main` queue being serviced which creates a bottle neck and an assumption that's often not true.